### PR TITLE
nautilus: rgw: fix Multisite sync corruption

### PR DESCRIPTION
**Pull request by xiaoxichen**
https://tracker.ceph.com/issues/40145

### CODE REVIEW

**Comment by tchaikov** [comment]
```
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/googletest/googletest/include/gtest/gtest.h:1874,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/rgw/test_http_manager.cc:22:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/rgw/test_http_manager.cc: In member function 'virtual void HTTPManager_ReadTruncated_Test::TestBody()':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/rgw/test_http_manager.cc:47:58: error: no matching function for call to 'RGWHTTP::process(RGWHTTPClient*, const optional_yield::empty_t&)'
   EXPECT_EQ(-EAGAIN, RGWHTTP::process(&client, null_yield));
                                                          ^
/home/jenkins-build/build/workspace/ceph-pull-requests/src/googletest/googletest/include/gtest/gtest_pred_impl.h:77:52: note: in definition of macro 'GTEST_ASSERT_'
   if (const ::testing::AssertionResult gtest_ar = (expression)) \
                                                    ^~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/googletest/googletest/include/gtest/gtest_pred_impl.h:162:3: note: in expansion of macro 'GTEST_PRED_FORMAT2_'
   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_NONFATAL_FAILURE_)
   ^~~~~~~~~~~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/googletest/googletest/include/gtest/gtest.h:1923:3: note: in expansion of macro 'EXPECT_PRED_FORMAT2'
   EXPECT_PRED_FORMAT2(::testing::internal:: \
   ^~~~~~~~~~~~~~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/rgw/test_http_manager.cc:47:3: note: in expansion of macro 'EXPECT_EQ'
   EXPECT_EQ(-EAGAIN, RGWHTTP::process(&client, null_yield));
   ^~~~~~~~~
In file included from /home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_keystone.h:13,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_auth.h:15,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_common.h:1923,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_rados.h:17,
                 from /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/rgw/test_http_manager.cc:14:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_http_client.h:363:14: note: candidate: 'static int RGWHTTP::process(RGWHTTPClient*)'
   static int process(RGWHTTPClient *req);
              ^~~~~~~
/home/jenkins-build/build/workspace/ceph-pull-requests/src/rgw/rgw_http_client.h:363:14: note:   candidate expects 1 argument, 2 provided
src/test/rgw/CMakeFiles/unittest_http_manager.dir/build.make:62: recipe for target 'src/test/rgw/CMakeFiles/unittest_http_manager.dir/test_http_manager.cc.o' failed
make[3]: *** [src/test/rgw/CMakeFiles/unittest_http_manager.dir/test_http_manager.cc.o] Error 1
CMakeFiles/Makefile2:23246: recipe for target 'src/test/rgw/CMakeFiles/unittest_http_manager.dir/all' failed
```

**Comment by tchaikov** [comment]
@xiaoxichen the FTBFS seems genuine .

File: src/test/rgw/test_http_manager.cc [hunk 1]
```diff
@@ -16,8 +16,39 @@
...
+        "HTTP/1.1 200 OK\r\n"
+        "Content-Length: 1024\r\n"
+        "\r\n"
+        "short body";
+    boost::asio::write(socket, boost::asio::buffer(response));
+  }};
+  const auto url = std::string{"http://127.0.0.1:"} + std::to_string(acceptor.local_endpoint().port());
+
+  RGWHTTPClient client{g_ceph_context, "GET", url};
+  EXPECT_EQ(-EAGAIN, RGWHTTP::process(&client, null_yield));
```
Comment (cbodley) [inline src/test/rgw/test_http_manager.cc]:
```diff
-  EXPECT_EQ(-EAGAIN, RGWHTTP::process(&client, null_yield));
+  EXPECT_EQ(-EAGAIN, RGWHTTP::process(&client));
```

**Comment by smithfarm** [comment]
@xiaoxichen nit: it should be "Conflicts:", not "Changes:", in the commit message - just for uniformity (automated tools might grep the commit messages for the string "Conflicts:")

**Comment by xiaoxichen** [comment]
@smithfarm I was snuggle which word to use....actually it is not a "conflicts" as the commit can cherry-pick cleanly.  But the signature of  RGWHTTP::process changed between nautilus and master
https://github.com/ceph/ceph/blob/nautilus/src/rgw/rgw_http_client.cc#L1213
which we have to change the cherry-pick code to match.

**Comment by yuriw** [comment]
[wip-yuri3-testing-2019-06-14-1440-nautilus](https://trello.com/c/217ldlvH/642-wip-yuri3-testing-2019-06-14-1440-nautilus)

**Comment by yuriw** [comment]
[wip-yuri3-testing-2019-06-17-1508-nautilus](https://trello.com/c/1nC2eLGN/645-wip-yuri3-testing-2019-06-17-1508-nautilus)

**Pull request merged**